api/src/main/kotlin/gropius/schema/mutation/UserMutations.kt

```diff
@@ -0,0 +1,213 @@
+        input: CreateIMSPermissionInput, dfe: DataFetchingEnvironment
+    ): IMSPermission {
+        return imsPermissionService.createIMSPermission(dfe.gropiusAuthorizationContext, input)
+    }
+
+    @GraphQLDescription(
+        """Updates a IMSPermission, requires ADMIN on all IMSs the permission currently affects.
+        Ensures that after the update, all affected IMSs still have a permission which grants ADMIN to at least
+        one user.
+        """
+    )
+    @AutoPayloadType("The updated IMSPermission")
+    suspend fun updateIMSPermission(
+        @GraphQLDescription("Defines which IMSPermission to update and how to update it")
+        input: UpdateIMSPermissionInput, dfe: DataFetchingEnvironment
+    ): IMSPermission {
+        return imsPermissionService.updateNodePermission(dfe.gropiusAuthorizationContext, input)
+    }
+
+    @GraphQLDescription(
+        """Deletes a IMSPermission, requires ADMIN on all IMSs the permission currently affects.
+        Ensures that after the permission is deleted, all previously affected IMSs still have a permission which 
+        grants ADMIN to at least one user.
+        """
+    )
+    @AutoPayloadType("The id of the deleted IMSPermission")
+    suspend fun deleteIMSPermission(
+        @GraphQLDescription("Defines which IMSPermission to delete")
+        input: DeleteNodeInput, dfe: DataFetchingEnvironment
+    ): ID {
+        imsPermissionService.deleteNodePermission(dfe.gropiusAuthorizationContext, input)
+        return input.id
+    }
+
+    @GraphQLDescription("Creates a new GlobalPermission, requires that the user is an admin")
+    @AutoPayloadType("The created GlobalPermission")
+    suspend fun createGlobalPermission(
+        @GraphQLDescription("Defines the created GlobalPermission")
+        input: CreateGlobalPermissionInput, dfe: DataFetchingEnvironment
+    ): GlobalPermission {
+        return globalPermissionService.createGlobalPermission(dfe.gropiusAuthorizationContext, input)
+    }
+
+    @GraphQLDescription("Updates a GlobalPermission, requires that the user is an admin")
+    @AutoPayloadType("The updated GlobalPermission")
+    suspend fun updateGlobalPermission(
+        @GraphQLDescription("Defines which GlobalPermission to update and how to update it")
+        input: UpdateGlobalPermissionInput, dfe: DataFetchingEnvironment
+    ): GlobalPermission {
+        return globalPermissionService.updateGlobalPermission(dfe.gropiusAuthorizationContext, input)
+    }
+
+    @GraphQLDescription("Deletes a GlobalPermission, requires that the user is an admin")
+    @AutoPayloadType("The id of the deleted GlobalPermission")
+    suspend fun deleteGlobalPermission(
+        @GraphQLDescription("Defines which GlobalPermission to delete")
+        input: DeleteNodeInput, dfe: DataFetchingEnvironment
+    ): ID {
+        globalPermissionService.deleteGlobalPermission(dfe.gropiusAuthorizationContext, input)
+        return input.id
+    }
+
+}
```

Comment:
Note for file:

I always found our naming convention of naming a node that represents the relation between users, components and a string "permission" because intuetively for me the string is the permission and the node should be something like a "granted permission on node" but lets leave it like it is

Reply:
currently we name both the container and the entry permission  
I also don't like this, however I did not find a better name.  
If you want to change any of those names, we can discuss this in the next sync meeting, as this would be a domain model change, and then update it in a separate PR

Reply:
Naah. Leave it. It't not grreat but every better name would make the class names so long, that they only fit on yout ultrawide